tests/rules_test.py

```diff
@@ -474,9 +475,32 @@ def test_new_term():
     with mock.patch('elastalert.ruletypes.Elasticsearch') as mock_es:
         mock_es.return_value = mock.Mock()
         mock_es.return_value.search.return_value = mock_res
+        call_args = []
+
+        # search is called with a mutable dict containing timestamps, this is required to test
+        def record_args(*args, **kwargs):
+            call_args.append((copy.deepcopy(args), copy.deepcopy(kwargs)))
+            return mock_res
+
+        mock_es.return_value.search.side_effect = record_args
         rule = NewTermsRule(rules)
 
-        assert rule.es.search.call_count == 2
+    # 30 day default range, 1 day default step, times 2 fields
+    assert rule.es.search.call_count == 60
+
+    # Assert that all calls have the proper ordering of time ranges
+    old_ts = '2010-01-01T00:00:00Z'
+    old_field = ''
+    for call in call_args:
+        field = call[1]['body']['aggs']['filtered']['aggs']['values']['terms']['field']
```

Comment:
These calls make Demeter sad: https://en.wikipedia.org/wiki/Law_of_Demeter

However, as always, we've found no better way to do this. Consider breaking these calls up into individual classes if you care to clean these up and make them easier to understand.